text/0000-json-getters.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: JSON Viewpoint Getters
+- Start Date: 2021-11-18
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+Introduce "getters" into the JSON package. Specifically access methods for
+`data` in `JsonDoc`, `JsonObject` and `JsonArray`.
+
+# Motivation
+
+Currently the interaction between the JSON API and the Pony reference
+capability type system makes it difficult to work with JSON documents that need
+to be mutated after they have been created.
+
+By introducing getter methods it will be possible to leverage viewpoint
+adaptation to enable mutations when the caller site holds a mutable reference,
+but to enforce immutability when the calling alias is a `val` or `box`.
+
+Additionally, a caller with an `iso` reference will be able to perform
+mutations to the document and then recover and `iso` reference afterwards.
+
+# Detailed design
+
+The implementation consists of using the `apply()` style access to introducing
+three getters, into `JsonDoc`, `JsonObject` and `JsonArray` respectively:
+
+- document data access: `fun apply(): this->JsonType! => ...`,
+- object data access: `fun apply(): this->Array[JsonType]! => ...`,
+- array data access: `fun apply(): this->Map[String, JsonType]! => ...`.
```

Comment:
Nit: these `!`s are not necessary here

Reply:
Fixed in the RFC. I still need to update the code PR.